pmd/legend-ruleset.xml

```diff
@@ -0,0 +1,50 @@
+<ruleset xmlns="http://pmd.sourceforge.net/ruleset/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         name="Default Maven PMD Plugin Ruleset"
+         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+    <description>The default ruleset used by the Maven PMD Plugin, when no other ruleset is specified. It contains the
+        rules of the old (pre PMD 6.0.0) rulesets java-basic, java-empty, java-imports, java-unnecessary,
+        java-unusedcode. This ruleset might be used as a starting point for an own customized ruleset [0]. [0]
+        https://pmd.github.io/latest/pmd_userdocs_making_rulesets.html
+    </description>
+    <rule ref="category/java/bestpractices.xml/AvoidUsingHardCodedIP"/>
+    <rule ref="category/java/bestpractices.xml/CheckResultSet"/>
+    <rule ref="category/java/bestpractices.xml/PrimitiveWrapperInstantiation"/>
+    <rule ref="category/java/bestpractices.xml/UnusedFormalParameter"/>
+    <rule ref="category/java/bestpractices.xml/UnusedLocalVariable"/>
+    <rule ref="category/java/bestpractices.xml/UnusedPrivateField"/>
+    <rule ref="category/java/bestpractices.xml/UnusedPrivateMethod"/>
+    <rule ref="category/java/codestyle.xml/EmptyControlStatement"/>
+    <rule ref="category/java/codestyle.xml/ExtendsObject"/>
+    <rule ref="category/java/codestyle.xml/ForLoopShouldBeWhileLoop"/>
+    <rule ref="category/java/codestyle.xml/TooManyStaticImports"/>
+    <rule ref="category/java/codestyle.xml/UnnecessaryFullyQualifiedName"/>
+    <rule ref="category/java/codestyle.xml/UnnecessaryImport"/>
+    <rule ref="category/java/codestyle.xml/UnnecessaryModifier"/>
+    <rule ref="category/java/codestyle.xml/UnnecessaryReturn"/>
+    <rule ref="category/java/codestyle.xml/UnnecessarySemicolon"/>
+    <rule ref="category/java/codestyle.xml/UselessQualifiedThis"/>
+    <rule ref="category/java/design.xml/CollapsibleIfStatements"/>
+    <rule ref="category/java/design.xml/SimplifiedTernary"/>
+    <rule ref="category/java/design.xml/UselessOverridingMethod"/>
+    <rule ref="category/java/errorprone.xml/AvoidBranchingStatementAsLastInLoop"/>
+    <rule ref="category/java/errorprone.xml/AvoidDecimalLiteralsInBigDecimalConstructor"/>
+    <rule ref="category/java/errorprone.xml/AvoidMultipleUnaryOperators"/>
+    <rule ref="category/java/errorprone.xml/AvoidUsingOctalValues"/>
+    <rule ref="category/java/errorprone.xml/BrokenNullCheck"/>
+    <rule ref="category/java/errorprone.xml/CheckSkipResult"/>
+    <rule ref="category/java/errorprone.xml/ClassCastExceptionWithToArray"/>
+    <rule ref="category/java/errorprone.xml/DontUseFloatTypeForLoopIndices"/>
+    <rule ref="category/java/errorprone.xml/EmptyCatchBlock"/>
```

Comment:
Whether EmptyCatchBlock is an appropriate rule depends on what PMD considers to be an "empty" catch block. Sometimes it is appropriate to catch and ignore an exception, though in such a case it's good to have a comment in the catch block to explain why. So I think it's fine to have this rule if "empty" means "only whitespace", but otherwise it's probably best to leave it out.

Reply:
we have multiple places in FlatData where we use empty catch blocks, for example (https://github.com/finos/legend-engine/blob/8fed6fa5a299d88a7273442865e05d6f285fbf26/legend-engine-xt-flatdata-shared/src/main/java/org/finos/legend/engine/external/format/flatdata/shared/driver/core/valueParser/DateParser.java#L57)

Will the plugin flag these? 

The logic here, is that given a list of date formats, try to parse a value using each one until one succeeds.

Reply:
I think it's preferrable to manually add  where a reasonable practice is violated on purpose by the developer, the solution is to instruct the linter to ignore the violation:

`
//NOPMD
`
is one option, the alternative is to use the annotations

```
// This will suppress UnusedLocalVariable warnings in this class
@SuppressWarnings("PMD.UnusedLocalVariable")
public class Bar {
    void bar() {
        int foo;
    }
}
```

Reply:
There are cases where an empty catch blocks are entirely appropriate. I'm not a fan of having to put annotations in the code base because PMD can't tell the difference between when empty catch blocks are appropriate and when they're in error. So my inclination is that we should leave this rule out for now.